src/services/user.ts

```diff
@@ -114,6 +118,46 @@ export class UserService implements IUserService {
     return this.signup(login, password);
   }
 
+  public async requestResetPassword(login: string): Promise<IUserResetPassword> {
+    this.validateEmail(login);
+    const userDb = await User.findOne({ login }).exec();
+    if (userDb) {
+      const secret = JWT_SECRET + userDb.passwordHash;
+      const paylod = {
+        email: login,
+        id: userDb._id,
+      };
+      const token = jwt.sign(paylod, secret, { expiresIn: '15m' });
+      const resetUrl = '?token=' + token + '&email=' + login;
+      sendResetPasswordEmail(login, resetUrl);
+      const resetLink = `${token} `;
+      return {
+        login,
+        link: resetLink,
+      };
+    }
+    // throw new BadRequestError('Email address is not in db.');
+  }
+  public async resetPassword(login: string, password: string, token: string): Promise<IUserResetPassword> {
+    try {
+      this.validatePassword(password);
+      const userDb = await User.findOne({ login }).exec();
+      const secret = JWT_SECRET + userDb.passwordHash;
+      jwt.verify(token, secret);
+      if (userDb) {
+        const newSalt = await bcrypt.genSalt(5);
+        const newPasswordHash = await bcrypt.hash(password, newSalt);
+        await User.updateOne({ _id: userDb._id }, { salt: newSalt });
```

Comment:
what's the point of making 2 updates of the same entity one after another? just do `{salt, passwordHash }`